src/traffic_server/InkAPI.cc

```diff
@@ -9751,6 +9751,50 @@ TSRegisterProtocolTag(const char *tag)
   return nullptr;
 }
 
+void
+TSHttpTxnRedoCacheLookup(TSHttpTxn txnp, const char *host, const int port, const char *path, const char *query)
```

Comment:
Question: Why not let this API take an URL ? Not a string, but an actual URL object.

Reply:
no idea, but that's a good suggestion, I can make that change.

Reply:
if this is going to be a public API, who do we expose the URL header? I guess that's the only reason I can think of to use each individual URL section instead of an object.

Reply:
are you referring to this URL object?

https://github.com/apache/trafficserver/blob/master/proxy/hdrs/URL.h

or this one?

https://github.com/apache/trafficserver/blob/master/include/tscpp/api/Url.h

Reply:
The URL would be passed in as a pair of `TSMBuffer` and `TSMLoc`. For example, see `TSUrlSchemeGet` which retrieves the scheme from a URL object.